### PR TITLE
Update docker base image & respect sonar-project.properties

### LINKED ISSUE #5: projectKey not read from sonar-project.properties
I run this action for a repository where I have defined a sonar-project.properties, in which I have defined: 

> `sonar.projectKey=my-project`

however during Action execution I can see the project key is always set to 'workspace' (overwriting results for any existing project with that key name):

> `INFO: Project key: workspace`

Other settings in sonar-project.properties seem to be read fine.
Is there a workaround for this behavior?

Thanks!

### PR DESCRIPTION
**Pull request by jankal**
### Description
As the title says, I updated the base image this image relies on so it now ueses SonarScanner v4.5. The image size might be accacted. I think, this is a non-issue bc GitHub handles pulling the image.  
I also added logic to detect a `sonar-project.properties` file in the current directory. When one is present the config settings from that file are used instead of settings from GH Actions or auto-generated settings (like `projectName` being the repo name).

### Related Issue
Fixes #5
Also updates Docker base image to Sonar-Scanner v4.5

### Motivation and Context
The upstream docker image already dates back one year. I also hope to get rid of the node:8 warnings in SonarQube.

### Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] ~New feature (non-breaking change which adds functionality)~
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### How Has This Been Tested?
I ran the action in one of my projects, where a `sonar-project.properties` file exists.

See [here](https://github.com/jankal/jankal.dev/runs/1573645255?check_suite_focus=true).

### CODE REVIEW

**Comment by dwisiswant0** [comment]
Noice, @jankal! :fire:
We'll go back as soon for review.

File: entrypoint.sh [hunk 1]
```diff
@@ -13,17 +13,26 @@ fi
...
-	-Dsonar.projectKey=${SONAR_PROJECTKEY} \
-	-Dsonar.projectName=${SONAR_PROJECTNAME} \
-	-Dsonar.projectVersion=${SONAR_PROJECTVERSION} \
-	-Dsonar.projectBaseDir=${INPUT_PROJECTBASEDIR} \
-	-Dsonar.login=${INPUT_LOGIN} \
-	-Dsonar.password=${INPUT_PASSWORD} \
-	-Dsonar.sources=. \
-	-Dsonar.sourceEncoding=UTF-8
+if [! -f "./sonar-project.properties"]
+then
```
[the remark this answers is not in the document]
Comment (dwisiswant0) [reply entrypoint.sh:18]: I also think what if we replace the base directory with the environment variable `GITHUB_WORKSPACE`? Like: `"${GITHUB_WORKSPACE}/sonar-project.properties"`.

**Review by dwisiswant0 — CHANGES_REQUESTED** [review]

**Review by dwisiswant0 — APPROVED** [review]

**Comment by dwisiswant0** [comment]
Hi, @jankal. Is c949545 passed test?

**Comment by jankal** [comment]
@dwisiswant0 thank you for checking back again. I found issues on line 17 of `entrypoint.sh` which I have to fix first. Do not merge.

**Comment by jankal** [comment]
@dwisiswant0 now I think it's all fixed up :)

**Pull request merged**